### ISSUE TITLE
False-positive `unneeded_concatenation` with single special-character name

### ISSUE THREAD

**Issue by bersbersbers** [r-lib/lintr#1327]
1a650ce9a934d30f142538212078247ef0360e7c warns about this code:
```r
x <- c("something$special" = "ordinary_stuff")
```

It does not warn about
```r
x <- c(something_ordinary = "ordinary_stuff")
```
but omitting the quotation marks is not possible when the name is not a valid identifier (or something like that).

Actual use case:
```r
devtools::check(env_vars = c("_R_CHECK_PKG_SIZES_THRESHOLD_" = "100"))
```

**Comment by AshesITR** [comment]
Thanks for reporting this!
